### PR TITLE
feat(stdlib): add experimental.preview function

### LINKED ISSUE #4719: Create experimental preview function for limiting the output
Create an experimental preview function. The preview function will limit the number of output series and the number of rows, but will not affect the sources so it will still run the entire query.

It would be the equivalent of something like the following:

```
|> slimit(n: 5)
|> limit(n: 5)
```

DOD:

- Preview function exists that limits the number of tables and number of rows per table
- Preview function has two parameters (table limit, row limit), maximum number of rows returned is (table limit * row limit). Parameters should have small defaults (i.e. 5)

### PR DESCRIPTION
**Pull request by jsternberg**
The preview function will limit the number of rows and the number of
tables that can be returned by a transformation. It will discard any
remaining tables/rows.

Fixes #4719.

### Done checklist
- [x] docs/SPEC.md updated
- [x] Test cases written

### CODE REVIEW

File: stdlib/experimental/experimental.flux [hunk 1]
```diff
@@ -1252,3 +1252,34 @@ builtin histogram : (
         bins: [float],
         ?normalize: bool,
     ) => stream[{T with _value: float, le: float}]
+
+// preview limits the number of rows and tables in the stream.
+// The group keys that are included are not deterministic and depends on the order
+// that the engine sends them.
```
Comment (sanderson) [inline stdlib/experimental/experimental.flux:1258]:
```suggestion
// preview limits the number of rows and tables in the stream.
//
// Included groups keys are not deterministic and depend on the order
// that the engine sends them.
```

File: stdlib/experimental/experimental.flux [hunk 2]
```diff
@@ -1252,3 +1252,34 @@ builtin histogram : (
...
+//
+// ## Parameters
+// - nrows: Maximum number of rows in each table.
```
Comment (sanderson) [inline stdlib/experimental/experimental.flux:1261]:
```suggestion
// - nrows: Maximum number of rows per table to return.
```

File: stdlib/experimental/experimental.flux [hunk 3]
```diff
@@ -1252,3 +1252,34 @@ builtin histogram : (
...
+//
+// - ntables: Maximum number of tables returned by the transformation.
```
Comment (sanderson) [inline stdlib/experimental/experimental.flux:1263]:
```suggestion
// - ntables: Maximum number of tables to return.
```

File: stdlib/experimental/experimental.flux [hunk 4]
```diff
@@ -1252,3 +1252,34 @@ builtin histogram : (
...
+//
+// ## Examples
+//
+// ### Preview data output
+// ```
+// import "experimental"
+// import "sampledata"
+//
+// sampledata.init()
+//   |> experimental.preview()
```
Comment (sanderson) [inline stdlib/experimental/experimental.flux:1275]:
```suggestion
// ```
// import "experimental"
// import "sampledata"
//
// sampledata.int()
//     |> experimental.preview()
// ```
```

File: stdlib/experimental/experimental.flux [hunk 5]
```diff
@@ -1252,3 +1252,34 @@ builtin histogram : (
...
+//
+// ## Metadata
+// introduced: next
```
Comment (sanderson) [inline stdlib/experimental/experimental.flux:1278]: @nathanielc Does case matter here? Does this need to be `NEXT`?

File: stdlib/experimental/experimental.flux (hunk 3 above)
Comment (sanderson) [reply stdlib/experimental/experimental.flux:1263]: Does this parameter have a default value? The example below suggests that it does.

File: stdlib/experimental/experimental.flux (hunk 2 above)
Comment (sanderson) [reply stdlib/experimental/experimental.flux:1261]: Does this parameter have a default value? The example below suggests that it does.

**Review by sanderson — CHANGES_REQUESTED** [review]

File: stdlib/experimental/experimental.flux (hunk 5 above)
Comment (jsternberg) [reply stdlib/experimental/experimental.flux]: It looks like it's supposed to be `NEXT`. Fixed.

**Review by wolffcm — APPROVED** [review]
Looks reasonable.

File: stdlib/experimental/experimental.flux [hunk 6]
```diff
@@ -1252,3 +1252,33 @@ builtin histogram : (
         bins: [float],
         ?normalize: bool,
     ) => stream[{T with _value: float, le: float}]
+
+// preview limits the number of rows and tables in the stream.
+//
+// Included group keys are not deterministic and depends on the order
+// that the engine sends them.
```
Comment (sanderson) [inline stdlib/experimental/experimental.flux:1259]:
```suggestion
// The function is designed to quickly provide a data preview by not
// returning the entire result set.
// Included group keys are not deterministic and depends on the order
// that the engine sends them.
```

File: stdlib/experimental/experimental.flux [hunk 7]
```diff
@@ -1252,3 +1252,33 @@ builtin histogram : (
...
+//
+// ## Examples
+//
+// ### Preview data output
+// ```
+// import "experimental"
+// import "sampledata"
+//
+// sampledata.init()
+//     |> experimental.preview()
```
Comment (sanderson) [inline stdlib/experimental/experimental.flux:1277]: - Change `sampledata.init()` to `sampledata.int()`
- Passing parameters in the example will show exactly how the input tables are modified.
- Add input (`<`) and output (`>`) identifiers to generate the input and output examples.

```suggestion
// < sampledata.int()
// >     |> experimental.preview(nrows: 3, ntables: 1)
```

**Review by sanderson — APPROVED** [review]

**Pull request merged**